ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Handlers.hs

```diff
@@ -85,26 +89,24 @@ handlerRequestNext currentIntersection blockTree points =
   runWriterT $ do
     intersection <- lift $ readTVar currentIntersection
     trace $ "  last intersection is " ++ condense intersection
-    withOrigin headerIsOrigin (withHeader intersection) (coerce (header points))
+    withHeader intersection (coerce (header points))
   where
-    headerIsOrigin = pure Nothing
-
-    withHeader :: Point TestBlock -> Header TestBlock -> WriterT [String] (STM m) (Maybe RequestNext)
+    withHeader :: Point TestBlock -> WithOrigin (Header TestBlock) -> WriterT [String] (STM m) (Maybe RequestNext)
     withHeader intersection h =
       maybe noPathError (analysePath hp) (BT.findPath intersection hp blockTree)
       where
-        hp = AF.castPoint $ blockPoint h
+        hp = AF.castPoint $ withOrigin GenesisPoint blockPoint h
```

Comment:
I feel `withOrigin GenesisPoint blockPoint` should be a named helper near `blockPoint`. Maybe `blockPoint'` or something?

Reply:
hah, I just added that 😅 though `blockPoint` is in `ouroboros-network`, so I put it in this file